spring-observability-core/src/main/java/org/springframework/observability/tracing/listener/TracingRecordingListener.java

```diff
@@ -40,26 +41,37 @@ public TracingRecordingListener(Tracer tracer) {
 
 	@Override
 	public void onStart(IntervalRecording<TracingContext> intervalRecording) {
-		Span span = this.tracer.nextSpan().name(intervalRecording.getEvent().getName())
+		Span span = this.tracer.nextSpan().name(intervalRecording.getName())
 				.start(getStartTimeInMicros(intervalRecording));
-		intervalRecording.getContext().setSpanInScope(this.tracer.withSpan(span));
+		intervalRecording.getContext().setSpanAndScope(span, this.tracer.withSpan(span));
 	}
 
 	@Override
 	public void onStop(IntervalRecording<TracingContext> intervalRecording) {
-		Span span = this.tracer.currentSpan();
+		SpanAndScope spanAndScope = intervalRecording.getContext().getSpanAndScope();
+		if (spanAndScope == null) {
```

Comment:
Can this actually happen?
I think this would indicate a bug in the implementation of the recording, i.e.: onStart is not called.